### DISCUSSION TITLE
Bitbucket-pipelines support

### DISCUSSION THREAD

**Discussion by iiian** [nektos/act#1556] [Ideas]
I’ve got an organization that **I personally** want to use GitHub (which would make this project useful to my day-to-day), but the org is pretty adamant about staying with Bitbucket.

I’m wondering if you guys think the model of computation is suitable to accept Bitbucket-pipelines.yml file as an optional input, or perhaps a “run-for-pipelines” mode. I know that Bitbuckets’ pipelines has its own containers it can plug-in which may or may not clash with concepts already supported by GitHub actions…

I’m aware this functionality probably doesn’t exist, but if it’s a simple enough refactor to support it I’d be happy to help.

**Comment by mytchallb** [comment]
I would love something like Act for Bitbucket too!

**Comment by im7mortal** [comment]
I checked if it would be possible.

1. Created simple `bitbucket-pipelines.yml` 
2. Unmarshaled it from yaml and put values in  [model.Workflow](https://github.com/nektos/act/blob/7172fc31d152db62b4fdca159e7853e977a43f95/pkg/model/workflow.go#L18-L25)
3. Mapped event
4. It run perfectly.🚀


### Challenges.

Most cases will work out of the box if convert `bitbucket-pipelines.yml` to github actions format.
I checked if there importers. I found only [official github importer](https://github.com/github/gh-actions-importer). It's in C#. It also requires to login to Github.

### Conclusion. 

It will require dynamic importer otherwise it will run perfectly.
